src/components/activity-completion/completion-page-content.tsx

```diff
@@ -1,50 +1,165 @@
+import { refIdToAnswersQuestionId } from "../../utilities/embeddable-utils";
+import ccPlaceholderLogo from "../../assets/cc-placeholder.png";
 
 import "./completion-page-content.scss";
 
 interface IProps {
+  activity: Activity;
   activityName: string;
-  isActivityComplete: boolean;
   onPageChange: (page: number) => void;
   showStudentReport: boolean;
   thumbnailURL: string | null;
+  onOpenReport?: () => void;
+  sequence?: Sequence;
+  activityIndex?: number;
+  onActivityChange?: (activityNum: number) => void;
+  onShowSequence?: () => void;
 }
 
 export const CompletionPageContent: React.FC<IProps> = (props) => {
-  const { activityName, isActivityComplete, onPageChange, showStudentReport, thumbnailURL } = props;
+  const { activity, activityName, onPageChange, showStudentReport, thumbnailURL,
+    sequence, activityIndex, onActivityChange, onShowSequence } = props;
+  
+  const [answers, setAnswers] = useState<any>();
+  let activityCompletionArray: boolean[]=[];
+
   const handleExit = () => {
-    onPageChange(0);
+    if (sequence) { onShowSequence?.(); }
+    else { onPageChange(0); }
+  };
+  const handleNextActivity = () => {
+    onActivityChange?.(activityNum + 1);
+    onPageChange(0); //TODO: This should go to the student's saved state page for the next activity
   };
   const handleShowAnswers = () => {
     showReport();
   };
-  const quotedActivityName = "\"" + activityName + "\"";
-  const completionText = `${quotedActivityName} activity complete!`;
-  const progressText = isActivityComplete
-      ? `Congratulations! You have reached the end of ${quotedActivityName} and your work has been saved`
-      : `It looks like you haven't quite finished ${quotedActivityName} yet. The answers you've given have been saved.`;
-  const nextStepText = isActivityComplete
-      ? `You have completed ${quotedActivityName} and you may exit now.`
-      : `You haven't completed ${quotedActivityName} yet. You can go back to complete it, or you can exit.`;
+
+  const sequenceProgress = () => {
+    let activityStatus;
+    sequence?.activities.map((sequenceActivity)=>{
+      activityStatus = activityProgress(sequenceActivity);    
+      const sequenceActivityComplete = activityStatus.numAnswers === activityStatus.numQuestions;
+      const activityCompletionArrayCopy = [...activityCompletionArray];
+      activityCompletionArrayCopy?.push(sequenceActivityComplete);
+      activityCompletionArray = [...activityCompletionArrayCopy];
+    });             
+    if (activityCompletionArray?.length === sequence?.activities.length && !activityCompletionArray?.includes(false)) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const activityProgress = (kActivity: Activity) => {
+    let answerNum = 0, questionNum = 0, questionId: string;
+    let i = 0;
+    for (i = 0; i < kActivity.pages.length; i++) {
+      kActivity.pages[i].embeddables.map((embeddableWrapper: EmbeddableWrapper) => {
```

Comment:
maps are probably not appropriate here since you're not really mapping the array to something new.  You're really searching the array and iterating something external.